service/service.go

```diff
@@ -97,9 +98,11 @@ func newService(set *settings) (*service, error) {
 		return nil, fmt.Errorf("cannot build pipelines: %w", err)
 	}
 
-	// The process telemetry initialization requires the ballast size, which is available after the extensions are initialized.
-	if err = telemetry.RegisterProcessMetrics(srv.telemetryInitializer.ocRegistry, getBallastSize(srv.host)); err != nil {
-		return nil, fmt.Errorf("failed to register process metrics: %w", err)
+	if set.Config.Telemetry.Metrics.Level != configtelemetry.LevelNone && set.Config.Telemetry.Metrics.Address != "" {
```

Comment:
Should we also have a test that covers the scenario of the  metric level being > None but the address not being set?  Its a pretty silly scenario but the if statement is checking for it.